components/inputs/input-number.js

```diff
@@ -313,6 +350,56 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 
 	}
 
+	_handleInput(e) {
+		if (e.inputType === 'insertFromPaste') {
+			this._handleChange(e);
+		}
+		if (e.inputType !== 'insertText') {
+			this._hintType = HINT_TYPES.NONE;
+		}
+	}
+
+	_handleKeyPress(e) {
+		const key = e.key;
+		let prevent = false;
+		let hintType = HINT_TYPES.NONE;
+		const hasDecimal = e.target.value.indexOf(this._descriptor.symbols.decimal) > -1;
+		if (key === this._descriptor.symbols.negative) {
+			// negative symbol must be at the end
+			if (this._descriptor.patterns.decimal.negativePattern === '{number}-') {
```

Comment:
Maybe I'm blanking, but what use case would the negative symbol at the end be for?

Reply:
In Arabic and other RTL locales, negative numbers would look like this: `92-` instead of `-92`.